tests/tasks/toloka/test_client.py

```diff
@@ -0,0 +1,61 @@
+import inspect
+import prefect
+import pytest
+import requests_mock
+from prefect.tasks.toloka.utils import (
+    DEFAULT_TOLOKA_ENV,
+    DEFAULT_TOLOKA_SECRET_NAME,
+    with_toloka_client,
+)
+from toloka.client import TolokaClient
+
+
+DEFAULT_TOKEN = "some-token"
+
+OTHER_SECRET_NAME = "OTHER_SECRET"
+OTHER_TOKEN = "some-other-token"
+OTHER_ENV = "SANDBOX"
+
+
+@pytest.fixture
+def secrets_mock():
+    secrets = {
+        DEFAULT_TOLOKA_SECRET_NAME: DEFAULT_TOKEN,
+        OTHER_SECRET_NAME: OTHER_TOKEN,
+    }
+    with prefect.context(secrets=secrets):
+        yield
+
+
+class TestWithTolokaClient:
+    def test_signature(self, secrets_mock):
+        @with_toloka_client
+        def some_func(arg1, arg2, toloka_client):
+            ...
+
+        params = set(inspect.signature(some_func).parameters)
+        assert {"arg1", "arg2", "secret_name", "env"} == params, params
+
+    def test_new_toloka_client(self, secrets_mock):
+        @with_toloka_client
+        def some_func(expected_token, expected_env, toloka_client=None):
+            assert expected_token == toloka_client.token
+            assert TolokaClient.Environment[expected_env].value == toloka_client.url
+
+        some_func(DEFAULT_TOKEN, DEFAULT_TOLOKA_ENV)
+        some_func(OTHER_TOKEN, OTHER_ENV, secret_name=OTHER_SECRET_NAME, env=OTHER_ENV)
+
+
+def test_add_headers(secrets_mock):
+    @with_toloka_client
+    def make_request(toloka_client):
+        return toloka_client.get_pool("123")
+
+    with requests_mock.Mocker(real_http=False) as mocker:
```

Comment:
Hi, Alexander! Thanks for taking a look :)
Yes, I've seen the page about contribution (as well as other tasks in the library), so it was an intentional decision to write tasks as functions.

It seems that the "class" form works fine when you have some arguments that can be declared before executing a flow. For example, these can be a bucket name and key when using AWS S3 tasks or a container and blob_name when using Azure Blob Storage tasks.

But as for our tasks, there's actually no arguments to reuse. Most of them operate with entities, that are going to be created during DAG's execution, so one can't instantiate a class with them and reuse it later.

I can rewrite tasks as classes, but their usages will be followed with empty parentheses at each.
For example, instead of:
```
pool = create_pool(...)
tasks = create_tasks(data, pool)
open_pool(pool)
```
it turns to:
```
pool = CreatePool()(...)
tasks = CreateTasks()(data, pool)
OpenPool()(pool)
```

Reply:
Yep, I can definitely see the advantage of your approach and I appreciate the way that you've organized the tasks you've created to be interoperable. I think one request that I would make would be to allow users to pass in their Toloka token so that they can use a different secrets backend if they choose to. If that's possible with your current structure then I'll be fine moving forward with the function-based paradigm for these tasks.

Reply:
@desertaxle Hi, Alexander! We have added the ability to pass a Toloka token as an input to our operations. So now users can use multiple accounts as follows:
```
token_b = PrefectSecret('OTHER_TOLOKA_TOKEN')

assignments_a = get_assignments()  # Use token from default TOLOKA_TOKEN secret.
assignments_b = get_assignments(token=token_b)  # Pass token from PrefectSecret.
```
We've also replaced `requests_mock` library to `responses` in our tests.